### PR TITLE
Split config into public and authenticated endpoint

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
For the `Login` page the `loginTypes` and `landingPageUrl` properties will be moved to the public `/login-config.json` endpoint. All other configuration that is not required for unauthenticated access is moved under `/api/config`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other developer
The frontend config is now split into an authenticated (`/api/config`) and unauthenticated endpoint (`/login-config.json`) 
```

### CODE REVIEW

File: backend/test/acceptance/__snapshots__/config.spec.js.snap [hunk 1]
```diff
@@ -1,5 +1,30 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`config should reject unauthenticated requests 1`] = `Array []`;
+
+exports[`config should reject unauthenticated requests 2`] = `
+Object {
+  "code": 401,
+  "details": Object {
+    "name": "UnauthorizedError",
+    "stack": "UnauthorizedError: No authorization token was found
```
Comment (holgerkoser) [inline backend/test/acceptance/__snapshots__/config.spec.js.snap:10]: A stack trace in a snap is not a good idea

File: backend/test/acceptance/config.spec.js [hunk 2]
```diff
@@ -36,4 +50,15 @@ describe('config', function () {
 
     expect(res.body).toMatchSnapshot()
   })
+
+  it('should reject unauthenticated requests', async function () {
```
Comment (holgerkoser) [inline backend/test/acceptance/config.spec.js:54]: We do not need this test. The middleware of `/api` is already tested in an own acceptance test. Test only the good case.

File: frontend/src/router/guards.js [hunk 3]
```diff
@@ -6,13 +6,17 @@
...
+  getConfiguration,
+  getPublicConfiguration
+} from '@/utils/api'
 
 export default function createGuards (store, userManager, localStorage) {
   return {
     beforeEach: [
       setLoading(store, true),
-      ensureConfigurationLoaded(store),
+      ensureConfigurationLoaded(store, userManager),
```
Comment (holgerkoser) [inline frontend/src/router/guards.js:19]: We can move this down if we have a static asset for public config

File: charts/gardener-dashboard/values.yaml [hunk 4]
```diff
@@ -101,8 +101,11 @@ oidc:
   #  # clientSecret is the public client secret use by kubelogin and all users
   #  clientSecret: ~
 
-frontendConfig:
+# frontendConfigPublic is the configuration that is accessible without authentication
+frontendConfigPublic:
```
Comment (holgerkoser) [inline charts/gardener-dashboard/values.yaml:105]: I would keep the `landingPageUrl` in `frontendConfig` and handle what is public in the tmeplate for now.

File: backend/test/acceptance/config.spec.js (hunk 2 above)
Comment (petersutter) [reply backend/test/acceptance/config.spec.js:54]: ✅

File: charts/gardener-dashboard/values.yaml (hunk 4 above)
Comment (petersutter) [reply charts/gardener-dashboard/values.yaml:105]: ✅

File: backend/test/acceptance/__snapshots__/config.spec.js.snap (hunk 1 above)
Comment (petersutter) [reply backend/test/acceptance/__snapshots__/config.spec.js.snap]: ✅

File: frontend/src/router/guards.js (hunk 3 above)
Comment (petersutter) [reply frontend/src/router/guards.js]: ✅

File: backend/lib/app.js [hunk 5]
```diff
@@ -57,12 +57,11 @@ app.use(helmet.hsts())
 app.use(noCache(['/js', '/css', '/fonts', '/img', '/static']))
 app.use('/auth', auth.router)
 app.use('/webhook', githubWebhook.router)
-app.use(['/api', '/config.json'], compression({
+app.use(['/api'], compression({
```
Comment (holgerkoser) [inline backend/lib/app.js:60]:
```suggestion
app.use('/api', compression({
```

File: frontend/src/router/guards.js [hunk 6]
```diff
@@ -36,10 +36,14 @@ function setLoading (store, value) {
 
 function ensureConfigurationLoaded (store) {
   return async function (to, from, next) {
+    const meta = to.meta || {}
```
Comment (holgerkoser) [inline frontend/src/router/guards.js:39]: If we move `ensureConfigurationLoaded` inside of `ensureDateLoaded` https://github.com/gardener/dashboard/blob/42fca41d290705e4bbe608376166f3bb743c34ae/frontend/src/router/guards.js#L101 we don't need this. And I think now we can consider config as data.

File: frontend/src/views/Login.vue [hunk 7]
```diff
@@ -122,6 +126,10 @@ export default {
     })
   },
   methods: {
+    async getLoginConfiguration () {
+      const { data: cfg } = await getLoginConfiguration()
+      assign(this.cfg, cfg)
```
Comment (holgerkoser) [inline frontend/src/views/Login.vue:131]:
```suggestion
      Object.assign(this.cfg, cfg)
```

File: frontend/src/views/Login.vue [hunk 8]
```diff
@@ -164,7 +172,8 @@ export default {
       this.$snotify.error(message, 'Login Error', config)
     }
   },
-  mounted () {
+  async mounted () {
```
Comment (holgerkoser) [inline frontend/src/views/Login.vue:175]: Why not in created?

File: frontend/src/router/guards.js (hunk 6 above)
Comment (petersutter) [reply frontend/src/router/guards.js]: ✅

File: frontend/src/views/Login.vue (hunk 8 above)
Comment (petersutter) [reply frontend/src/views/Login.vue]: ✅

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**